front/tests/utils/trainSchedule-utils.ts

```diff
@@ -0,0 +1,51 @@
+import type { APIResponse, APIRequestContext } from '@playwright/test';
+
+import { postApiRequest, getApiContext } from './api-setup';
+
```

Comment:
It's just a suggestion, but I think it clearer to extract the handleApiResponse function : 

```suggestion
// Helper function to handle API error responses
async function handleApiResponse(response: APIResponse, errorMessage: string): Promise<void> {
  if (!response.ok()) {
    throw new Error(`${errorMessage}: ${response.status()} ${response.statusText()}`);
  }
}
```